pkg/serverless/frequency.go

```diff
@@ -0,0 +1,73 @@
+package serverless
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/serverless/flush"
+)
+
+const maxInvocationsStored = 50
+
+// StoreInvocationTime stores the given invocation time in the list of previous
+// invocations. It is used to compute the invocation frequency of the current function.
+// It is automatically removing entries when too much have been already stored (more than maxInvocationsStored).
+// When trying to store a new point, if it is older than the last one stored, it is ignored.
+// Returns if the point has been stored.
+func (d *Daemon) StoreInvocationTime(t time.Time) bool {
+	// ignore points less recent than the last stored one
+	if len(d.lastInvocations) > 0 && d.lastInvocations[len(d.lastInvocations)-1].After(t) {
+		return false
+	}
+
+	// remove when too much/old entries
+	d.lastInvocations = append(d.lastInvocations, t)
+	if len(d.lastInvocations) > maxInvocationsStored {
```

Comment:
Alternatively, we could use an [exponential moving average](https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average). That way we don't allocate any memory, and we weight for the most recent invocations. It's usually pretty simple, eg `newAverage = lastAverage + averagingFactor * ( newValue - lastAverage)`. The averaging factor is a constant we pick, usually 0<x<1. The larger it is, the more weight recent invocations have.

Reply:
About the memory usage, for completeness, with 50 values, we are speaking of `24 + 50*8=424 bytes`.
About the lookback, I don't mind reducing its size since I feel like the adaptive flush will mainly be useful for function quickly reaching their interval of invocation.

First I'll push a change with less values for now, let's say 10, and see in a second time if switching to a moving average is better, that's a good idea.